code.js

```diff
@@ -1,3 +1,46 @@
-function dijkstra(graph, sourceNode) {
-    return [];
+function dijkstra(graph, sourceNode)  {
+
+    if(graph.size === 0){
+        return null;
+    }
+
+    var distances = {};
+    var nodes = Object.keys(graph);
+    var visited = new Set(Object.keys(graph));
```

Comment:
This looks like it's the unvisited nodes?

Reply:
Still relevant -- if this keeps track of the *un*visited nodes, don't call it visited.